### PR TITLE
feat: implement user session management with jwt and database persistence

### PR DESCRIPTION
**Pull request by Tha-Orakkle**
## Description
This PR introduces a comprehensive user session management system. It allows users to see the different devices signed into their account -including their locations- and provides a way to log out from all or specific devices.

## Features
The key enhancements include:

Active Session Tracking: Each issued JWT refresh token is stored in the database with associated metadata such as user ID, device details, IP address, expires_at, and location.

Session Revocation: Users can view and selectively revoke active sessions, enabling logout from specific devices without affecting other sessions.

Enhanced Security: By maintaining a centralized session store, compromised tokens can be individually invalidated, reducing potential security risks.

Regular DB Management: Every Sunday midnight, expired and revoked tokens are removed from the db.

​
## Related Issue (Link to issue ticket)
https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/1061

## Motivation and Context

Implementing session management enhances user experience by providing control over active sessions and improves security by allowing precise token invalidation. This aligns with best practices for authentication systems, addressing the current limitations where JWTs are stateless and lack centralized management.

## How Has This Been Tested?

The following testing strategies were employed

- **Integration Tests**: Simulated user authentication workflows to verify that session data is accurately recorded and managed.

- **Manual Testing**: Performed using Postman to validate endpoints for session management, including edge cases like attempting to revoke already invalidated tokens.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/auth.py [hunk 1]
```diff
@@ -245,6 +258,15 @@ def login(request: Request, login_request: LoginRequest, background_tasks: Backg
     # Generate access and refresh tokens
     access_token = user_service.create_access_token(user_id=user.id)
     refresh_token = user_service.create_refresh_token(user_id=user.id)
+    expires = dt.datetime.now(dt.timezone.utc) + (dt.timedelta(
+        days=settings.JWT_REFRESH_EXPIRY) - dt.timedelta(seconds=1)
+    )
+    session_schema: SessionCreate = get_session_schema_data(
```
Comment (samuelogboye) [inline api/v1/routes/auth.py]: inside this method get_session_schema_data, you are calling get_ip_location which is calling an external API
This will increase latency during login

**Review by samuelogboye — CHANGES_REQUESTED** [review]
Ensure comprehensive testing of the session management system, especially edge cases like concurrent session access and token revocation.
Then make sure all test are passing(currently the CI is failing)

**Comment by Tha-Orakkle** [comment]
I have refactored the get_session_schema_data to function asynchronously to prevent latency.
Tessts have been created for the different endpoints
